src/main/kotlin/com/autonomousapps/extension/IssueHandler.kt

```diff
@@ -65,66 +69,104 @@ open class IssueHandler @Inject constructor(objects: ObjectFactory) {
+    return issuesFor(projectPath) { it.anyIssue }
+  }
+
+  internal fun unusedDependenciesIssueFor(projectPath: String): List<Provider<Behavior>> {
+    return issuesFor(projectPath) { it.unusedDependenciesIssue }
+  }
+
+  internal fun usedTransitiveDependenciesIssueFor(projectPath: String): List<Provider<Behavior>> {
+    return issuesFor(projectPath) { it.usedTransitiveDependenciesIssue }
   }
 
-  internal fun unusedDependenciesIssueFor(path: String): Provider<Behavior> {
-    val global = all.unusedDependenciesIssue
-    val proj = projects.findByName(path)?.unusedDependenciesIssue
-    return overlay(global, proj)
+  internal fun incorrectConfigurationIssueFor(projectPath: String): List<Provider<Behavior>> {
+    return issuesFor(projectPath) { it.incorrectConfigurationIssue }
   }
 
-  internal fun usedTransitiveDependenciesIssueFor(path: String): Provider<Behavior> {
-    val global = all.usedTransitiveDependenciesIssue
-    val proj = projects.findByName(path)?.usedTransitiveDependenciesIssue
-    return overlay(global, proj)
+  internal fun compileOnlyIssueFor(projectPath: String): List<Provider<Behavior>> {
+    return issuesFor(projectPath) { it.compileOnlyIssue }
   }
 
-  internal fun incorrectConfigurationIssueFor(path: String): Provider<Behavior> {
-    val global = all.incorrectConfigurationIssue
-    val proj = projects.findByName(path)?.incorrectConfigurationIssue
-    return overlay(global, proj)
+  internal fun runtimeOnlyIssueFor(projectPath: String): List<Provider<Behavior>> {
+    return issuesFor(projectPath) { it.runtimeOnlyIssue }
   }
 
-  internal fun compileOnlyIssueFor(path: String): Provider<Behavior> {
-    val global = all.compileOnlyIssue
-    val proj = projects.findByName(path)?.compileOnlyIssue
-    return overlay(global, proj)
+  internal fun unusedAnnotationProcessorsIssueFor(projectPath: String): List<Provider<Behavior>> {
+    return issuesFor(projectPath) { it.unusedAnnotationProcessorsIssue }
   }
 
-  internal fun runtimeOnlyIssueFor(path: String): Provider<Behavior> {
-    val global = all.runtimeOnlyIssue
-    val proj = projects.findByName(path)?.runtimeOnlyIssue
-    return overlay(global, proj)
+  internal fun redundantPluginsIssueFor(projectPath: String): Provider<Behavior> {
+    return overlay(all.redundantPluginsIssue, projects.findByName(projectPath)?.redundantPluginsIssue)
   }
 
-  internal fun unusedAnnotationProcessorsIssueFor(path: String): Provider<Behavior> {
-    val global = all.unusedAnnotationProcessorsIssue
-    val proj = projects.findByName(path)?.unusedAnnotationProcessorsIssue
-    return overlay(global, proj)
+  internal fun moduleStructureIssueFor(projectPath: String): Provider<Behavior> {
+    return overlay(all.moduleStructureIssue, projects.findByName(projectPath)?.moduleStructureIssue)
+  }
+
+  private fun issuesFor(projectPath: String, mapper: (ProjectIssueHandler) -> Issue): List<Provider<Behavior>> {
+    val projectHandler = projects.findByName(projectPath)
```

Comment:
The algorithm here is so ugly.

Reply:
🤷 if it works. I think as you said, it could be a topic for `2.x` of the plugin to rework this whole part of the DSL so that it also becomes easier to process.

Still I think the PR as ist has now has value.